### PR TITLE
Fix 2019 agi logic so that clients who did non filer arent asked for agi

### PR DESCRIPTION
**Pull request by bytheway875**

### CODE REVIEW

File: spec/forms/ctc/spouse_filed2019_form_spec.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      it "is not valid" do
+        expect(
+            described_class.new(intake, {})
+        ).not_to be_valid
+      end
+    end
+
+    context "when spouse_filed_2019 is not in the set" do
+      let(:filed_2019) { "on_the_moon" }
+      it "is not valid" do
```
Comment (alaroia-cfa) [inline spec/forms/ctc/spouse_filed2019_form_spec.rb:28]: Consider `it "is valid at the form level but invalid at the model level" do`

to explain this behavior.

Not a blocker.

Comment (alaroia-cfa) [reply spec/forms/ctc/spouse_filed2019_form_spec.rb:28]: If the model has testing for this validation, then consider removing this test.

File: spec/forms/ctc/spouse_filed2019_form_spec.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        expect {
+          described_class.new(intake, params).save
+        }.to change(intake, :spouse_filed_2019).from("unfilled").to("did_not_file")
+        .and not_change(intake, :spouse_prior_year_agi_amount)
+      end
+    end
+
+    context "filed_non_filer_separate" do
+      let(:filed_2019) { "filed_non_filer_separate" }
+      it "updates spouse filed 2019, does not set agi" do
```
Comment (alaroia-cfa) [inline spec/forms/ctc/spouse_filed2019_form_spec.rb:79]: Consider

`      it "updates spouse filed 2019, does set agi" do`

I think it **does** set the spouse AGI but the test description states that it does not set the AGI.

File: spec/forms/ctc/spouse_filed2019_form_spec.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        expect {
+          described_class.new(intake, params).save
+        }.to change(intake, :spouse_filed_2019).from("unfilled").to("filed_non_filer_joint")
+         .and change(intake, :spouse_prior_year_agi_amount).from(nil).to(1)
+      end
+    end
+
+    context "filed_full_separate" do
+      let(:filed_2019) { "filed_full_separate" }
+      it "updates spouse filed 2019, sets AGI to 1" do
```
Comment (alaroia-cfa) [inline spec/forms/ctc/spouse_filed2019_form_spec.rb:59]: Consider: `it "updates spouse filed 2019 and does not modify AGI"`.

I think this description is not accurate the way it is written - I can see in the test it does not change an AGI, but the `it ...` line says it does change an AGI.

File: spec/forms/ctc/spouse_filed2019_form_spec.rb [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+          described_class.new(intake, params).save
+        }.to raise_error ArgumentError
+      end
+    end
+  end
+
+  context "save" do
+    context "filed_full_joint" do
+      let(:filed_2019) { "filed_full_joint" }
+      it "updates spouse filed 2019, does not set agi" do
```
Comment (alaroia-cfa) [inline spec/forms/ctc/spouse_filed2019_form_spec.rb:39]: These tests have a lot of boilerplate, which makes it hard to know what's unique about each test case. Consider a `before ... do` with `described_class.new(intake, params).save`.

I know that doing that makes it hard to use the `expect { ... }.to change(...)` syntax. I would rather see:

```
      let(:filed_2019) { "filed_full_joint" }
      it "updates spouse filed 2019, does not set agi" do
         intake.reload
         expect(intake.spouse_filed_2019).to eq("filed_full_joint")
         expect(intake.spouse_prior_year_agi_amount).to eq(nil)
      end
```

The tests are all so similar that I wish it weren't necessary to write the "it ..." description name at all. Maybe this is a good use for some of these ideas:

https://stackoverflow.com/questions/8754582/best-way-to-organize-tests-in-rspec-that-have-a-combinations-of-factor

Anyway, for now, please do fix the test description names, or come up with a better way to organize these! :)

**Review by alaroia-cfa — APPROVED** [review]
It's OK as it is! I have some quibbles about test descriptions which I would love to see addressed. They're not strictly blockers but I hope you will consider making those changes. Fine to merge as-is or with those changes.

File: spec/forms/ctc/spouse_filed2019_form_spec.rb (hunk 3 above)
Comment (bytheway875) [reply spec/forms/ctc/spouse_filed2019_form_spec.rb:59]: oh yeah, i just copy and pasted all of these tests. test descriptions are annoying. will change

**Comment by bytheway875** [comment]
made test changes to descriptions, but not doing any further refactoring

**Pull request merged**